apps/client/src/app/services/internet-identity.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthClient } from '@dfinity/auth-client';
+import { OAuthResponse } from '@ghostfolio/common/interfaces';
+import { EMPTY } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InternetIdentityService {
+  public constructor(private http: HttpClient) {}
+
+  public async login(): Promise<OAuthResponse> {
+    const authClient = await AuthClient.create();
```

Comment:
FYI per default `AuthClient` observes misc events and automatically sign-out after a couple of minutes (like 10) if user is not active.

If you wish to deactivate this behavior (as I do i Papyrs), you can create it as following:

```
const createAuthClient = (): Promise<AuthClient> =>
  AuthClient.create({
    idleOptions: {
      disableIdle: true,
      disableDefaultIdleCallback: true
    }
  });
```

Reply:
Not sure if I need it 🤔  I only use the `AuthClient` to get the `principalId`, then I generate a custom `jwt` with expiration time.

Reply:
In that case I would disable the checker - i.e. you don't need to attach these listener if in any case you are not going to check if the delegation of internet identity expires or not.

Reply:
> In that case I would disable the checker - i.e. you don't need to attach these listener if in any case you are not going to check if the delegation of internet identity expires or not.

✅  Done